.github/workflows/publish.yml

```diff
@@ -0,0 +1,94 @@
+# Publishes built jars to distribution platforms
+name: Publish
+
+on:
+  push:
+    tags:
+      - 'v*.*.*' # any semver tag, e.g. v1.2.3
+
+# if a second commit is pushed quickly after the first, cancel the first one's build
+concurrency:
+  group: publish-${{github.head_ref}}
+  cancel-in-progress: true
+
+env:
+  # link to the changelog with a format code for the version
+  CHANGELOG_LOCATION: "Changelog is available [here](https://github.com/${{github.repository}}/releases/tag/${{github.ref_name}})"
+  # type of release
+  RELEASE_TYPE: "beta"
```

Comment:
In case that GTCEu becomes release this will need to be changed. Also this will prevent us from having option to produce more then one type of release at same time.

Reply:
I'm not sure how to go about finding a solution for this. Currently we only publish from `master` as a `beta` release to curseforge. I don't envision publishing alpha releases other than on GitHub, and additionally I think the repository would transfer over to a full `release` state when the time comes, and then not publish betas on other platforms after.

Reply:
Is their an option to pass an optionally "Release Type" string when running the action? With the new PR testing actions, we might not be publishing betas as often, but it still could be the case where we want to publish a beta after going full release.